server/routes/index.ts

```diff
@@ -63,15 +53,42 @@ export default function routes(service: Services): Router {
     const { user } = res.locals
     const systemToken = await hmppsAuthClient.getSystemClientToken(user.username)
     const incidentReportingApi = new IncidentReportingApi(systemToken)
+    const manageUsersApiClient = new ManageUsersApiClient()
 
     // No authorisation at this point, no data shown in production
     if (config.environment === 'prod') {
       throw new NotFound()
     }
 
     const report = await incidentReportingApi.getReportWithDetailsById(id)
+    const reportedBy = (await manageUsersApiClient.getNamedUser(systemToken, report.reportedBy))?.name
+
+    res.render('pages/report', { report, reportedBy })
+  })
+
+  get('/incidents', async (req, res, next) => {
+    const { user } = res.locals
+    const systemToken = await hmppsAuthClient.getSystemClientToken(user.username)
+    const incidentReportingApi = new IncidentReportingApi(systemToken)
+    const manageUsersApiClient = new ManageUsersApiClient()
+
+    const incidents = (
+      await incidentReportingApi.getEvents({
+        // prisonId: user.activeCaseLoadId,
+        // eventDateFrom: new Date('2024-07-30'),
+        // eventDateUntil: new Date('2024-07-30'),
+        // sort: ['eventDateAndTime,ASC'],
+      })
+    )?.content
+
+    const inputUsernames = []
+    for (const incident of incidents) {
+      inputUsernames.push(incident.modifiedBy)
+    }
+
+    const nameMappings = await makeUsernameLookup(manageUsersApiClient, systemToken, inputUsernames)
```

Comment:
I do wonder if we could have a method in the `manageUsersApiClient` that gets the list of usernames and returns the mapping for you (but I'd map to the whole `User` to make the client more generic in case we need to use other information in other places)

Similar to what I've done for getting the prisoners names:

https://github.com/ministryofjustice/hmpps-incident-reporting/pull/135/files#diff-a5514f8aaeb918d8c1f5e7d5855b6179fcbb73ac46dcef3784aaea9e4c5c3d80R15-R28

That would make the Manage Users client more user-friendly and you don't need an utility function nor pass the client around as an argument :)

Reply:
So do you mean to effectively move the function I created into the `manageUsersApiClient` and keep the whole `user` and not just the name? Just want to clarify as there currently isn't an API call to retrieve multiple users at once, so it would still need to be the same API call done for each username. 
Happy to look to implement this, just want to clarify first.

Reply:
yeah exactly that! makes it more reusable

Reply:
I have done this. It is now as `getUsers` in the `userService`.